airflow/models.py

```diff
@@ -4285,6 +4285,27 @@ def set(cls, key, value, serialize_json=False, session=None):
         session.add(Variable(key=key, val=stored_value))
         session.flush()
 
+    @classmethod
+    @provide_session
+    def delete(cls, key, session=None):
+        try:
+            session.query(cls).filter(cls.key == key).delete()
+            session.commit()
```

Comment:
This seems suspect - why do we want to commit the session here - shouldn't the caller be in change of the session instead?